Fix getTotalStockInCarts not working and speed up

RVXD opened the pull request:
Using getClassName to identify Product or ProductVariation is not working because of namespacing.
Speed up TotalStockInCarts by using single Query instead of loop on all carts. 
Tends to be slow if many carts are added.

RVXD committed:
Fix getTotalStockInCarts not working and speed up

Using getClassName to identify Product or ProductVariation is not working because of namespacing.
Speed up TotalStockInCarts by using single Query instead of loop on all carts. 
Tends to be slow if many carts are added.

RVXD committed:
Remove commented line

wilr commented:
@RVXD Can you review the failure in https://travis-ci.org/github/silvershop/silvershop-stock/jobs/712854648

RVXD committed:
Fix to get Test working

The test is using $item->getStockBaseIdentifier().
Test could also use $item->getClassName() instead.
Changed back the getStockBaseIdentifier method to return the ClassName.

RVXD committed:
hasAvailableStock caching, less queries

RVXD committed:
remove debug line

wilr commented on src/Extensions/ProductStockExtension.php:
```diff
@@ -173,6 +183,7 @@ public function hasAvailableStock($require = 1)
      */
     public function getTotalStockInCarts()
     {
+        return 0;
```
@RVXD Hmm?

RVXD committed:
remove test line

RVXD committed:
remove caching line

RVXD committed:
Fix test - wrong column name

VariationOrderItem has_one is ProductVariation, so column should be ProductVariationID instead of VariationID.

RVXD commented:
Tested this locally with PHPUnit 5.7.27:  
vendor/bin/phpunit vendor/silvershop/stock/tests

On clean install with composer requirements:
```
"silverstripe/recipe-plugin": "^1.2",
"silverstripe/recipe-cms": "4.x@stable",
"silverstripe-themes/simple": "~3.2.0",
"silvershop/stock": "dev-patch-1 as 2.x-dev"
```

Not sure why tests fail on travis. @wilr Can you assist?

Result:
```
vendor/bin/phpunit vendor/silvershop/stock/tests
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.
.....                                                               5 / 5 (100%)
Time: 3.39 seconds, Memory: 42.00MB
OK (5 tests, 10 assertions)
```
